darts/utils/data/training_dataset.py

```diff
@@ -63,6 +70,102 @@ def __len__(self) -> int:
     def __getitem__(self, idx: int):
         pass
 
+    def _memory_indexer(self,
+                        ts_idx: int,
+                        ts_target: TimeSeries,
+                        shift: int,
+                        input_chunk_length: int,
+                        output_chunk_length: int,
+                        end_of_output_idx: int,
+                        ts_covariate: TimeSeries,
+                        cov_type: Optional[bool] = None) -> SampleIndexType:
```

Comment:
[Nitpicking] It seems the type hint is not good here. Also, I think you could use an Enum instead of two strings.

Reply:
True!